flake.nix

```
@@ -0,0 +1,25 @@
+{
+  description = "Universal Package Manager";
+
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, flake-utils }:
+    flake-utils.lib.eachDefaultSystem
+      (system:
+        let
+          pkgs = nixpkgs.legacyPackages.${system};
+          upm = pkgs.callPackage ./upm.nix {
+            rev = if self ? rev then "0.1.0-${builtins.substring 0 7 self.rev}" else "0.1.0-dirty";
```

```suggestion
            rev = if self ? rev then "0.0.0-${builtins.substring 0 7 self.rev}" else "0.0.0-dirty";
```
Small nitpick, we could try keeping this as "version 0" to clearly state that our versions depend on the revision and not semantic versioning? I will be creating an issue on the nix repo about this hard "must start with a digit" requirement.

---

Sure thing!